terra-env/variables.tf

```diff
@@ -41,11 +41,26 @@ variable "owner" {
 locals {
   owner = var.owner == "" ? terraform.workspace : var.owner
 }
+
+
+#
+# Preview environment Vars
+#
 variable "preview" {
   type        = bool
-  description = "Preview environment flag"
+  description = "Preview environment flag. Set to true if creating a preview environment."
   default     = false
 }
+variable "preview_shared" {
```

Comment:
Are all 4 combinations of the values of `preview` and `preview_shared` valid? I'd find it helpful to list their meanings explicitly.
It looks like the most common checks against these are
- `!var.preview && !var.preview_shared` for resources that should exist only for "full" environments
- `!var.preview_shared` for "running" environments  that actually run services (open to better terminology).
- `!var.preview` for potentially "shared" resources.
Consider whether we can define positive booleans variables to check instead of always examining negations, which add a little cognitive overhead while reading.

I think we want an enum, but that isn't supported by terraform. I'm not sure if special string values are better or worse?

Reply:
Thanks, this is very good feedback! I will play around to see if I can get some more intuitive syntax.